airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -219,21 +234,38 @@ def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Ite
 
         stream_state = stream_state or {}
         channels_stream = Channels(authenticator=self.authenticator)
+
         if self.cursor_field in stream_state:
             # Since new messages can be posted to threads continuously after the parent message has been posted, we get messages from the latest date
-            # found in the state minus 7 days to pick up any new messages in threads.
+            # found in the state minus X days to pick up any new messages in threads.
             # If there is state always use lookback
             messages_start_date = pendulum.from_timestamp(stream_state[self.cursor_field]) - self.messages_lookback_window
         else:
             # If there is no state i.e: this is the first sync then there is no use for lookback, just get messages from the default start date
             messages_start_date = pendulum.from_timestamp(self._start_ts)
+
         messages_stream = ChannelMessages(authenticator=self.authenticator, default_start_date=messages_start_date)
+
         for message_chunk in messages_stream.stream_slices(stream_state={self.cursor_field: messages_start_date.timestamp()}):
             self.logger.info(f"Syncing replies {message_chunk}")
+
             for channel in channels_stream.read_records(sync_mode=SyncMode.full_refresh):
                 message_chunk["channel"] = channel["id"]
+
                 for message in messages_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=message_chunk):
-                    yield {"channel": channel["id"], self.cursor_field: message[self.primary_key]}
+                    yield {"channel": channel["id"], self.sub_primary_key_2: message[self.sub_primary_key_2]}
+
+    def read_records(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+        """
+        Filtering already readed records for incremental sync. Copied state value to X after the last sync
+        to really 100% make sure no one can edit the state during the run.
+        """
+
+        initial_state = copy.deepcopy(stream_state)
+
+        for record in super().read_records(stream_state=stream_state, **kwargs):
+            if record.get(self.cursor_field, 0) >= initial_state.get(self.cursor_field, 0):
+                yield record
```

Comment:
Is it executes on incremental sync only ? If no, maybe we need to check if user wants to perform incremental or full_refresh

Reply:
Seems like `stream_state` will be None on full_refresh, so don't mind